File: src/main/java/com/yao/blog/ApplicationWebXml.java
package com.yao.blog;

import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;


/**
 * ApplicationWebXml
 */
public class ApplicationWebXml extends SpringBootServletInitializer {

    @Override
    protected SpringApplicationBuilder configure(SpringApplicationBuilder builder) {
        return builder.sources(BlogApp.class);
    }

}

File: src/main/kotlin/com/yao/blog/ApplicationWebXml.kt
package com.yao.blog;

import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;

public class ApplicationWebXml: SpringBootServletInitializer() {

    protected override fun configure(builder: SpringApplicationBuilder): SpringApplicationBuilder {
        return builder.sources(BlogApp::class.java);
    }

}